front/app/containers/Admin/reporting/components/ReportBuilder/Widgets/ChartWidgets/DemographicsWidget/StackedBarChart/parse.ts

```diff
@@ -0,0 +1,117 @@
+import { Multiloc } from 'typings';
+
+import { DemographicsResponse } from 'api/graph_data_units/responseTypes';
+
+import { Localize } from 'hooks/useLocalize';
+
+import { DEFAULT_CATEGORICAL_COLORS } from 'components/admin/Graphs/styling';
+
+import { binAge } from 'utils/dataUtils';
+import { roundPercentages } from 'utils/math';
+
+export const parseResponse = (
+  response: DemographicsResponse,
+  localize: Localize,
+  blankLabel: string
+) => {
+  const { options } = response.data.attributes;
+
+  // Birthyear is the only custom field we support here
```

Comment:
But according to this line https://github.com/CitizenLabDotCo/citizenlab/pull/7691/files#diff-a9780ab4aec49fc29626b1f4b145d8593868639af2eb1876e8bea3604791de78R37, we fetch all number fields, don't we? And for all of them `options` is `undefined`.

Reply:
Before we pass the custom fields to the select, we filter out all number fields that are not `birthyear` (there is no way to tell the BE to do this already atm): https://github.com/CitizenLabDotCo/citizenlab/pull/7691/files#diff-a9780ab4aec49fc29626b1f4b145d8593868639af2eb1876e8bea3604791de78R69

Reply:
Ah, got it; thanks for the link!